### PR TITLE
Print stack trace information for `AbstractThrowableAssert.doesNotThr…

### PR DESCRIPTION
**Pull request by mkobit**
…owAnyException()`

closes https://github.com/joel-costigliola/assertj-core/issues/1118

#### Check List:

* Fixes https://github.com/joel-costigliola/assertj-core/issues/1118
* Unit tests : YES
* Javadoc with a code example (API only) : YES / NO / NA

### CODE REVIEW

File: src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java [hunk 1]
```diff
@@ -39,17 +40,14 @@ public void can_invoke_late_assertion_on_assertThatCode() {
...
-    thrown.expectAssertionError(format("[Test] %n" +
-                                       "Expecting code not to raise a throwable but caught a%n" +
-                                       "  <java.lang.Exception>%n" +
-                                       "with message :%n" +
-                                       "  \"boom\""));
+    thrown.expectAssertionError(ShouldNotHaveThrown.shouldNotHaveThrown(exception));
 
     // When
-    assertThatCode(boom).as("Test").doesNotThrowAnyException();
+    assertThatCode(boom).doesNotThrowAnyException();
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java]: why removing `as("Test")` ? 
we want to make sure that the error allows a description set with `as`

File: src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java [hunk 2]
```diff
@@ -39,17 +40,14 @@ public void can_invoke_late_assertion_on_assertThatCode() {
...
+    Exception exception = new Exception("boom");
+    ThrowingCallable boom = raisingException(exception);
 
     // Expect
-    thrown.expectAssertionError(format("[Test] %n" +
-                                       "Expecting code not to raise a throwable but caught a%n" +
-                                       "  <java.lang.Exception>%n" +
-                                       "with message :%n" +
-                                       "  \"boom\""));
+    thrown.expectAssertionError(ShouldNotHaveThrown.shouldNotHaveThrown(exception));
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java]: static import `shouldNotHaveThrown`

Comment (joel-costigliola) [reply src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java]: add a separate test to verify the error message content as in https://github.com/joel-costigliola/assertj-core/blob/08c276844c55e74b5773248e7c5c9b6a741a4b4d/src/test/java/org/assertj/core/error/ShouldBeBefore_create_Test.java

Comment (mkobit) [reply src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java]: I wasn't sure of the best way to test both the description with `as()` and also testing the "stacktrace" way without making the test brittle - I'll take another look at the stacktrace case and maybe I can just assert that the message begins with the description in the `as()` test case? WDYT?

Comment (joel-costigliola) [reply src/test/java/org/assertj/core/api/Assertions_assertThatCode_Test.java]: `startsWith` and `contains` String assertions should do the job.

**Comment by joel-costigliola** [comment]
Integrated thanks @mkobit !

**Pull request closed without merging**